proposals/052-cryptography-leg-1.md

```diff
@@ -0,0 +1,329 @@
+
+- botan-bindings
+- botan-low
+- botan
+
+The full project has a large scope, and will require significant time to complete. More advanced concepts such as typeclasses may be delivered in a future leg focused on generalizing.
+
+### Expected project leg duration
+
+We estimate that it will take 3 months for this leg, to achieve our goal of developing and publishing these libraries. This estimate is based on the significant amount of work already completed, the length of time in which it was accomplished, and the remaining amount of work.
+
+Several work items (strictness, memory safety, C++ shims for missing FFI functionality) are difficult to estimate, and so time has been allocated for them in the estimate, but their delivery is considered optional, and may roll over into a future leg as necessary.
+
+### Intermediate deliverables
+
+The following are intermediate deliverables intended to keep the community apprised of any progress that is made, and any challenges that arise, and to seek community feedback:
+
+- Frequent (2-3x weekly) activity update on devlog
+- Weekly major update that includes github push
+- Monthly report that summarizes what specific items have been worked on, what items will be worked on, and what challenges have arisen, if any.
+
+### Deliverables
+
+The following are the intended deliverables of this leg of the proposal:
+
+- `botan-bindings`
+- `botan-low`
+- `botan`
+
+Each library should be beta- or / production-ready, and submitted to hackage as a candidate or accepted package. 
+
+Each library should have:
+
+- Documentation
+- Tutorials and examples
+- Tracked issues
+- Unit tests
+
+A project website will also be created.
+
+The official repo will be owned by the Haskell Cryptography Group, and another member of the Haskell Cryptography Group will be selected to hold backup keys and permissions to the website, server, and official repositories as needed, in order to give project members access and to reduce any conflicts about ownership or maintenance.
+
+## Budget
+
+The first leg of this proposal presents a minimum budget of $7000 USD per month, for one full-time engineer, for a duration of 3 months, for a total of $21,000 USD. This budget is based on cost-of-living and industry experience, and will cover housing, food, bills, taxes, and other life necessities for one engineer, as well as any project necessities such website and server hosting. This budget is roughly equivalent to $40 / hr or $84k / yr at full-time of 40 hrs / wk. Industry rates for an engineer of the necessary skill are on average considerably higher, and so we consider this budget to be reasonable. The exact legal contract / arrangement is left to the Haskell Foundation.
+
+The question of continued funding will be revisited for each future leg of this proposal, wherein the budget and direction of future work will be decided again for each leg.
```

Comment:
I'm a bit confused by the structure of legs here. Does accepting this proposal mean recommending that the whole series should be funded, or that just the specific work in this document should be funded?

Perhaps the proposal can be for a single project that delivers value even if not followed up on, rather than as the first step in a long series of tasks?

Reply:
And then of course there can be follow-up proposals for future development. Basically, we only have enough info to evaluate this phase, so it would be good to be explicit that this is what is being recommended or not.

Reply:
> I'm a bit confused by the structure of legs here. Does accepting this proposal mean recommending that the whole series should be funded, or that just the specific work in this document should be funded?

I wasn't sure on what approach the Haskell Foundation would find acceptable, or how long-term of a project it would be interested in funding. It is a considerable investment, and I am a relative newcomer to the community, and I felt it best to offer a short-term roadmap to build trust rather than require a leap of faith.

> Perhaps the proposal can be for a single project that delivers value even if not followed up on, rather than as the first step in a long series of tasks?

This is my intent (and why I have narrowed scope). I am interested in proposing a whole series, but what that later work looks will take more time to refine and would get in the way of accomplishing the immediate focus / goal of specifically developing the `botan` bindings. This proposal is specifically for the Botan libraries. If it is successful, then by the time this leg is finished, I will have a much better description of what the next leg would involve.
 
> And then of course there can be follow-up proposals for future development. Basically, we only have enough info to evaluate this phase, so it would be good to be explicit that this is what is being recommended or not.

This is accurate - I have outlined this phase sufficiently, but I will need more time to develop a proposal for future / advanced work.